bridges/snowbridge/primitives/router/src/outbound/mod.rs

```diff
@@ -69,13 +69,15 @@ where
 			return Err(SendError::NotApplicable)
 		}
 
-		let dest = destination.take().ok_or(SendError::MissingArgument)?;
+		// Cloning destination to avoid modifying the value so subsequent exporters can use it.
+		let dest = destination.clone().take().ok_or(SendError::MissingArgument)?;
 		if dest != Here {
 			log::trace!(target: "xcm::ethereum_blob_exporter", "skipped due to unmatched remote destination {dest:?}.");
 			return Err(SendError::NotApplicable)
 		}
 
-		let (local_net, local_sub) = universal_source
+		// Cloning universal_source to avoid modifying the value so subsequent exporters can use it.
+		let (local_net, local_sub) = universal_source.clone()
```

Comment:
I would maybe also change this:
```
log::error!(target: "xcm::ethereum_blob_exporter", "could not get global consensus from universal source '{universal_source:?}'.");
				SendError::Unroutable
```
to:

```
log::error!(target: "xcm::ethereum_blob_exporter", "could not get global consensus from universal source '{universal_source:?}'.");
				SendError::NotApplicable
```
to give a chance other routers in the tuple.

Reply:
and other suggestion change MissingArgument->NotApplicable:
```
		let para_id = match local_sub.as_slice() {
			[Parachain(para_id)] => *para_id,
			_ => {
				log::error!(target: "xcm::ethereum_blob_exporter", "could not get parachain id from universal source '{local_sub:?}'.");
                                // TODO: why not `NotApplicable`?
				return Err(SendError::MissingArgument)
			},
		};
```

```
	/// A needed argument is `None` when it should be `Some`.
	MissingArgument,
```

Reply:
and one more, I would move `message.take()` as a last step after all the validations pass.
And also `Unroutable` could be `NotApplicable` to give other router a chance.

```
		let message = message.take().ok_or_else(|| {
			log::error!(target: "xcm::ethereum_blob_exporter", "xcm message not provided.");
			SendError::MissingArgument
		})?;
		let source_location = Location::new(1, local_sub.clone());
		let agent_id = match AgentHashedDescription::convert_location(&source_location) {
			Some(id) => id,
			None => {
				log::error!(target: "xcm::ethereum_blob_exporter", "unroutable due to not being able to create agent id. '{source_location:?}'");
				return Err(SendError::Unroutable)
			},
		};
```